src/source/geojson_source.js

```diff
@@ -134,9 +140,18 @@ class GeoJSONSource extends Evented implements Source {
                 this.fire('error', {error: err});
                 return;
             }
+
+            const data = {};
+            data.dataType = 'source';
+            data.sourceDataType = 'metadata';
```

Comment:
Nit: 
```js
const data = {
    dataType: 'source',
    sourceDataType: 'metadata'
}
```

(also below)

Reply:
That's actually what I wrote originally, but it didn't survive `test-flow`. I think this has to stay this way because of the subsequent assignment of `data.resourceTiming`.  Flow complains about

```js
const data = { x: 'y' };
data.i = 'j';
```

but not

```js
const data = {};
data.x = 'y';
data.i = 'j';
```

My knowledge of Flow is pretty rudimentary so if there's a better way to accommodate this case please lmk.

Reply:
That's actually what I wrote originally, but it didn't survive `test-flow`. I think this has to stay this way because of the subsequent potential assignment of `data.resourceTiming`.  Flow complains about

```js
const data = { x: 'y' };
data.i = 'j';
```

but not

```js
const data = {};
data.x = 'y';
data.i = 'j';
```

My knowledge of Flow is pretty rudimentary so if there's a better way to accommodate this case please lmk.

Reply:
Ah, gotcha.  Does the following avoid the flow error?

```js
const data: MapDataType = {
    dataType: 'source',
    sourceDataType: 'metadata'
}
```

(you might have to import `MapDataType` from `ui/events.js`)